plotter/image.go

```diff
@@ -27,6 +28,11 @@ type Image struct {
 // (xmin, ymin) and (xmax, ymax) points given in the data space.
 // The img will be scaled to fit inside the rectangle.
 func NewImage(img image.Image, xmin, ymin, xmax, ymax float64) *Image {
+	if src, ok := img.(*image.Uniform); ok {
+		uni := image.NewRGBA(image.Rect(0, 0, int(xmax-xmin+0.5), int(ymax-ymin+0.5)))
+		draw.Draw(uni, uni.Bounds(), src, image.Point{}, draw.Src)
+		img = uni
+	}
```

Comment:
```suggestion
	if src, ok := img.(*image.Uniform); ok {
		img = cropped{src, image.Rect(0, 0, int(xmax-xmin+0.5), int(ymax-ymin+0.5))}
	}
```
with this after the `NewImage` body means we don't need to do any copying,
```
type cropped struct {
	*image.Uniform
	rect image.Rectangle
}

func (c cropped) Bounds() image.Rectangle { return c.rect }
```
It also means the confusing rename goes away.